google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -140,21 +140,22 @@ public DocumentReference document(@Nonnull String documentPath) {
   @Override
   public ApiFuture<List<DocumentSnapshot>> getAll(
       @Nonnull DocumentReference... documentReferences) {
-    return this.getAll(documentReferences, null, null);
+    return this.getAll(documentReferences, null, null, null);
   }
 
   @Nonnull
   @Override
   public ApiFuture<List<DocumentSnapshot>> getAll(
       @Nonnull DocumentReference[] documentReferences, @Nullable FieldMask fieldMask) {
-    return this.getAll(documentReferences, fieldMask, null);
+    return this.getAll(documentReferences, fieldMask, null, null);
   }
 
   /** Internal getAll() method that accepts an optional transaction id. */
   ApiFuture<List<DocumentSnapshot>> getAll(
       final DocumentReference[] documentReferences,
       @Nullable FieldMask fieldMask,
-      @Nullable ByteString transactionId) {
+      @Nullable ByteString transactionId,
+      @Nullable final ApiStreamObserver apiStreamObserver) {
```

Comment:
@ajaaym We would like to get your opinion on this approach:

Here getAll() the existing method returns ApiFuture<List<DocumentSnapshot>>, Now we have to provide support for ApiStreamObserver.

Abhinav has updated this very method to have a two-point return (I mean If apiStreamObserver is null then it returns future but it is not null then it uses that observer.

In my opinion, I think breaking this method into two is part(one with future & one with observer) a would be better but before suggesting new change we thought to confirm with you.